pkg/config/config.go

```diff
@@ -883,6 +883,9 @@ func InitConfig(config Config) {
 	// a more substantial refactor of autodiscovery is made to determine this automatically.
 	config.BindEnvAndSetDefault("logs_config.use_podman_logs", false)
 
+	// If set, the default docker path will be overridden.
```

Comment:
Since we're not documenting this option in `config_template.yaml`, I believe this will be the main/only documentation for this flag.

With that in mind, let's document this a bit more, maybe call out that on Docker's side this corresponds to the `data-root` option.

Can also link to the official docs, and maybe include `docker info -f '{{ .DockerRootDir}}'` as a potential debugging tool to ensure that this is set to a valid location?